.github/workflows/docs.yml

```diff
@@ -39,9 +39,17 @@ jobs:
       run: |
         sudo apt-get install pandoc
 
-    - name: Build docs
+    - name: Set build mode
       run: |
-        just docs
+        if [ ${{ contains(github.event.pull_request.labels.*.name, 'pr:docs-build-notebook') }} ]; then
+          DOCS_BUILD_MODE=skip
+        else
+          DOCS_BUILD_MODE=release
+        fi
+        echo "Running in '$DOCS_BUILD_MODE' mode"
```

Comment:
Keeping this as a separate step so that it's easy to look into the step's output for debugging purposes: should anything change in the GH workflow logic and this step break, we can see it immediately in the logs, instead of having it hidden among the thousands of lines from the actual build step